code/modules/chemistry/tools/robospray.dm

```diff
@@ -6,41 +6,64 @@
 	item_state = "syringe_0"
 	icon_state = "hypoborg"
 	var/inj_amount = 5
-	var/picker = 1
 	var/sound/sound_inject = 'sound/items/hypo.ogg'
-	var/botreagents = list(
+	var/max_fill_amount = 25
+	var/list/botreagents = list(
+		"charcoal" = 25,
 		"epinephrine" = 25,
-		"salbutamol" = 25,
 		"mannitol" = 25,
-		"saline" = 25,
-		"charcoal" = 25,
-		"anti_rad" = 25
+		"anti_rad" = 25,
+		"salbutamol" = 25,
+		"saline" = 25
 	)
+	var/list/available_chems = null
 	var/currentreagent = "epinephrine"
-	var/propername = "Epinephrine"
+	var/propername = "epinephrine"
+	var/image/fluid_image
+	var/extra_refill = 0
 
 	hide_attack = 2
 	inventory_counter_enabled = 1
 
 	New()
 		..()
 		processing_items.Add(src)
+		src.update_icon()
 
 	disposing()
 		..()
 		processing_items.Remove(src)
 
+	proc/update_icon()
+		if (botreagents[currentreagent] >= 1)
+			if (!src.fluid_image)
+				src.fluid_image = image(src.icon, "hypoover", -1)
+			var/datum/reagent/R = reagents_cache[src.currentreagent]
+			src.fluid_image.color = rgb(R.fluid_r, R.fluid_g, R.fluid_b, 255)
+			src.UpdateOverlays(src.fluid_image, "fluid")
+		else
+			src.UpdateOverlays(null, "fluid")
+		src.inventory_counter.update_number(botreagents[currentreagent])
+		signal_event("icon_updated")
+
 	attack_self(mob/user as mob)
-		picker = picker % length(botreagents) + 1
-		currentreagent = botreagents[picker]
-		var/datum/reagent/temp_reagent = reagents_cache[currentreagent]
-		propername = temp_reagent.name
+		if (available_chems == null)
+			available_chems = list()
+			for (var/reagent in botreagents)
```

Comment:
```suggestion
			for (var/reagent in sortList(botreagents))
```

Reply:
this would sort the list as

1. potassium iodide
2. charcoal
3. epinephrine
4. mannitol
5. salbutamol
6. saline-glucose solution

Reply:
oooooh...
https://github.com/goonstation/goonstation/blob/d6d8daeb6a01347e2224eb548be063548217c1f8/code/mob/living/silicon/ai/camera_handling.dm#L123